swarm/network/simulation/kademlia_test.go

```diff
@@ -63,4 +85,47 @@ func TestWaitTillHealthy(t *testing.T) {
 			t.Fatal(err)
 		}
 	}
+
+	// now create a snapshot of this network
+	snap, err := sim.Net.Snapshot()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// create a control simulation
+	controlSim := New(simServiceMap)
+	defer controlSim.Close()
+
+	// load the snapshot into this control simulation
+	err = controlSim.Net.Load(snap)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// for each node...
+	nodeIDs := controlSim.UpNodeIDs()
```

Comment:
Why is this up-nodes? Shouldn't we use _all_ nodes? We expect all nodes to have been brought up in the first simulation, so none of them should be down after we load the simulation.

Reply:
At this point it is correct o use Up nodes - in the control simulation, we want to check all nodes which actually have been started and are running.

But, putting it into context, for this kind of use case there should be no difference. So I am not so sure. @janos what do you say?

Reply:
> At this point it is correct o use Up nodes - in the control simulation, we want to check all nodes which actually have been started and are running.

I would say we want to check that _all_ nodes _were_ started, _and_ they are healthy.

Reply:
This is indeed what we really want @nolash, so I added a check for that

Reply:
we want ids set by `AddXX` of the first sim